packages/alfa-rules/test/sia-r21/rule.spec.tsx

```diff
@@ -67,6 +67,18 @@ test("evaluates() fails an element with an invalid role and not included in the
   ]);
 });
 
+test("evaluates() fails an element with hidden role", async (t) => {
+  const target = h.attribute("role", "hidden");
+
+  const document = h.document([h("button", [target])]);
+
+  t.deepEqual(await evaluate(R21, { document }), [
+    failed(R21, target, {
+      1: Outcomes.HasNoValidRole,
+    }),
+  ]);
+});
+
```

Comment:
`hidden` is an attribute in itself, and it should make the rule inapplicable.
So, something like `<button role="invalid">` fails, but `<button hidden role="invalid">` (or `<button style="display: none" role="invalid">`) doesn't.

```suggestion
test("evaluates() fails an element with hidden role", async (t) => {
const button = <button hidden role="invalid">Hello</button>

  const document = h.document([button]);

  t.deepEqual(await evaluate(R21, { document }), [
    inapplicable(R21)
  ]);
});

```